File: VB_Classes/SkiaSharp.vb
Imports cv = OpenCvSharp
Imports sk = SkiaSharp
' https://docs.microsoft.com/en-us/samples/xamarin/xamarin-forms-samples/skiasharpforms-demos/
Public Class SkiaSharp_Hello
    Inherits VBparent
    Dim hello As New CS_Classes.SkiaSharp_Hello
    Public Sub New()
        initParent()
        task.desc = "Create a bitmap with SkiaSharp and display it"
    End Sub
    Public Sub Run()
        If task.intermediateReview = caller Then ocvb.intermediateObject = Me
        dst1 = hello.Run(src.CvtColor(cv.ColorConversionCodes.BGR2GRAY))
    End Sub
End Class

File: CS_Classes/SkiaSharp.cs
using SkiaSharp;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Media.Animation;
using cv = OpenCvSharp;
using sk = SkiaSharp;

// https://docs.microsoft.com/en-us/samples/xamarin/xamarin-forms-samples/skiasharpforms-demos/
namespace CS_Classes
{
    public class SkiaSharp_Hello
    {
        const string TEXT = "Hello, Bitmap!";
        SKBitmap helloBitmap;
        public void New()
        {
        }
        public cv.Mat Run(cv.Mat src)
        {
            // Create bitmap and draw on it
            using (SKPaint textPaint = new SKPaint { TextSize = 48 })
            {
                SKRect bounds = new SKRect();
                textPaint.MeasureText(TEXT, ref bounds);

                helloBitmap = new SKBitmap((int)bounds.Right, (int)bounds.Height);

                using (SKCanvas bitmapCanvas = new SKCanvas(helloBitmap))
                {
                    bitmapCanvas.Clear();
                    bitmapCanvas.DrawText(TEXT, 0, -bounds.Top, textPaint);
                }
            }
            return src;
        }
    }
}
